libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/areaChart/PluggableAreaChart.tsx

```diff
@@ -232,4 +244,56 @@ export class PluggableAreaChart extends PluggableBaseChart {
             stacks,
         };
     }
+
+    private getBucketItemsWithMultipleDates(referencePoint: IReferencePoint) {
+        const buckets = referencePoint?.buckets ?? [];
+        const measures = getMeasureItems(buckets);
+        const masterMeasures = filterOutDerivedMeasures(measures);
+
+        let views: IBucketItem[] = [];
+        let stacks: IBucketItem[] = getStackItems(buckets, [ATTRIBUTE, DATE]);
+        const allAttributes = getAllAttributeItemsWithPreference(buckets, [
+            BucketNames.LOCATION,
+            BucketNames.TREND,
+            BucketNames.VIEW,
+            BucketNames.ATTRIBUTES,
+            BucketNames.SEGMENT,
+            BucketNames.STACK,
+            BucketNames.COLUMNS,
+        ]);
+
+        const dateItem = getFistDateItem(buckets);
+        if (dateItem) {
+            // first available date is put to views (date has preference)
+            views = [dateItem];
+
+            const [nextAttribute] = allAttributes.filter((attribute) => attribute !== dateItem);
+            const isNextAttributeDate = isDateBucketItem(nextAttribute);
+            if (masterMeasures.length <= 1 && nextAttribute && isNextAttributeDate) {
+                // next attribute is date -> put to stacks
+                stacks = [nextAttribute];
+            } else if (masterMeasures.length <= 1 && nextAttribute && !stacks.length) {
+                // next attribute is not date and stacks are empty -> put to views
+                views = [...views, nextAttribute];
+            }
+        } else {
+            if (masterMeasures.length <= 1 && allAttributes.length > 0 && !stacks.length) {
+                // have more attributes and stacks empty -> fill views
+                views = allAttributes.slice(0, MAX_VIEW_COUNT);
+            } else if (masterMeasures.length <= 1 && allAttributes.length) {
+                // have more attributes and stacks non-empty -> keep only one attribute in view
+                views = getAllCategoriesAttributeItems(buckets).slice(0, 1);
+            } else {
+                // more measures -> prefer measures
+                views = getAllCategoriesAttributeItems(buckets).slice(0, 1);
+                stacks = [];
+            }
+        }
```

Comment:
please keep an eye going forward.. there may be functional patterns emerging when doing other visualizations; and you may be able to identify high level functions to wrap some of this stuff into. no need to go overboard now; however if there is possibility to make these

Reply:
I will keep an eye on it definitely. I was surprised how different the charts are in this. So far we've covered bar/column/line and now area, and in which there were some tiny differences in constructing the reference point which prevented the unification.